crates/sui-types/src/object.rs

```diff
@@ -640,6 +640,14 @@ pub enum ObjectRead {
     NotExists(ObjectID),
     Exists(ObjectRef, Object, Option<MoveStructLayout>),
     Deleted(ObjectRef),
+    /// The object exists but not found in this node (could be pruned)
+    ExistsButPastNotFound(ObjectID, SequenceNumber),
```

Comment:
I think `ExistsButVersionNotFound` is probably a better name.

Reply:
indeed, thanks for the suggestion